src/main/java/com/minecolonies/util/BlockUtils.java

```diff
@@ -109,138 +113,201 @@ public static boolean isWater(IBlockState iBlockState)
     }
 
     /**
-     * Gets an item from a block.
-     * @param block input block.
-     * @return output item.
+     * Mimics pick block.
+     * @param blockState the block and state we are creating an ItemStack for.
+     * @return ItemStack fromt the BlockState.
      */
-    public static Item getItemFromBlock(Block block)
+    public static ItemStack getItemStackFromBlockState(IBlockState blockState)
     {
-        Item item = null;
+        Item item = getItem(blockState);
 
-        if(block instanceof BlockBanner)
+        if (item == null)
         {
-            item = Items.banner;
+            return null;
         }
-        else if(block instanceof BlockBed)
+
+        Block block = blockState.getBlock();
+        if (item instanceof ItemBlock && !block.isFlowerPot())
+        {
+            block = Block.getBlockFromItem(item);
+        }
+
+        return new ItemStack(item, 1, getDamageValue(block, blockState));
+    }
+
+    private static Item getItem(IBlockState blockState)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 31, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)